src/main/java/com/google/gerrit/extensions/api/changes/Changes.java

```diff
@@ -74,7 +75,7 @@ ChangeApi id(String project, String branch, String id)
     public abstract List<ChangeInfo> get() throws RestApiException;
 
     public QueryRequest withQuery(String query) {
-      this.query = query;
+      this.query = StandardKeyEncoder.encode(query);
```

Comment:
I agree 1. is not an option then. 2. maybe is not that bad after all, because even with 1. you could argue that users might forgot to enable the encoding to override the default. I'll play around with some ideas a bit and push something new over the weekend.